feat: retrospect, comment 구현

ybkang1108 opened the pull request:
- retrospect 작성, 수정, 조회, 삭제 구현 closes #21 
- comment 작성, 수정, 조회, 삭제 구현 closes #22

ybkang1108 committed:
feat: 회고 작성, 수정 구현

- round의 application에 RetrospectService 추가
- RoundMemberRepository 추가

ybkang1108 committed:
feat: 회고 삭제, 조회 기능 구현

- 회고 삭제
- 개별 회고 조회
- 모든 회고 조회

기능 구현

ybkang1108 committed:
refactor: response 수정

api명세서에 맞게 response 수정

ybkang1108 committed:
feat: comment 작성, 수정, 조회, 삭제 구현 및 Controller 분리

- RoundController에서 Comment와 Retrospect 분리
- Comment를 save하기 위한 CustomRepository 추가
- 기능 구현

ybkang1108 committed:
feat: 댓글, 회고 기능 구현

- 댓글 생성, 수정, 삭제, 조회
- 회고 생성, 수정, 삭제, 조회

ybkang1108 committed:
fix: 예외 처리 추가

- service에서 throw CustomException 추가
- controller에 try catch 추가
- 예외처리 구체화 필요..

ybkang1108 committed:
chore: todo 추가

ybkang1108 committed:
refactor: try catch 제거

cmj7271 commented on src/main/java/com/gdgoc/study_group/comment/domain/Comment.java:
```diff
@@ -24,4 +30,19 @@ public class Comment {
+            .member(member)
+            .comment(comment)
+            .build();
+  }
```
작은 도메인은 allArg 열어도 될거 같아서,
필요하면 allArg 생성자로 변경해주세요

cmj7271 commented on src/main/java/com/gdgoc/study_group/comment/domain/Comment.java:
```diff
@@ -24,4 +30,19 @@ public class Comment {
+
+  public void update(CommentDTO comment) {
+    this.comment = comment.comment();
+  }
```
좀 더 유연하게 String 으로 받는게 좋아보입니다

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/api/CommentController.java:
Comment 디렉토리로 옮기는 게 좋아보입니다

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/dto/CommentDTO.java:
요거 제가 노션에 잘못 적은거 같은게, member_id 도 있어야할 거 같아요.
API 명세서에 반영할테니까 수정 부탁드립니다

ybkang1108 replied:
혹시 댓글 작성/수정도 member_id 필요한가요??

cmj7271 commented:
comment 관련 레포지토리, 서비스, api 를 comment 도메인으로 옮기는 편이 좋을 거 같습니다.
제가 리뷰 마저 하고 반영해서 커밋하겠습니다.

cmj7271 commented on src/main/java/com/gdgoc/study_group/roundMember/domain/RoundMember.java:
```diff
@@ -26,4 +31,19 @@ public class RoundMember {
 
   private String retrospect; // 후기 내용
+
+  public static RoundMember create(
+          Round round,
+          Member member,
+          String retrospect) {
+    return RoundMember.builder()
+            .round(round)
+            .member(member)
+            .retrospect(retrospect)
+            .build();
+  }
```
생성자가 필요하다고 판단되었다면, AllArg 생성자를 public 으로 풀어서 사용해주세요

cmj7271 commented on src/main/java/com/gdgoc/study_group/roundMember/domain/RoundMember.java:
```diff
@@ -26,4 +31,19 @@ public class RoundMember {
+            .build();
+  }
+
+  public void update(String retrospect) {
```
조금더 명확한 함수명이 필요해보입니다

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/api/RetrospectController.java:
회고 작성(studies/{study_id}/rounds/{round_id}/retrospects) 과 회고 수정(studies/{study_id}/rounds/{round_id}/retrospects) 에 대해 반환값으로 `round_member_id` 반환하기로 했는데,
사실상 사용할 일이 없을 것 같아 제거하는게 좋아보입니다. 명세서에서 제거했으니 반영해주시면 감사하겠습니다

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/dao/RoundCustomRepository.java:
저희 기존 레포 생성 방법이랑은 다르기도 하고,
comment 는 옮겨질 예정이니까 제거해도 될거 같아요
roundMember 저장은 roundRepository에서 `@Query` 문 사용하는 걸로 할께요

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/dao/RoundCustomRepositoryImpl.java:
RoundCustomRepository 와 같은 이유로 제거해도 될거 같습니다.

cmj7271 reviewed (changes requested):
리뷰 올렸습니다
추가로, repository 는 변동사항 좀 있을거 같아서 유심히 봐주세요

ybkang1108 committed:
refactor: round custom repository 제거

- custom repository 제거
- @Query를 사용하여 Round Member저장하도록 변경

ybkang1108 committed:
refactor: retrospect update 함수 이름 변경

ybkang1108 committed:
refactor: 회고 작성/삭제 반환값 제거

ybkang1108 committed:
refactor: comment 이동

- allargs -> public, create제거
- comment를 round에서 comment로 이동
- comment dto: response와 request 분리
- comment response에는 member_id 추가

ybkang1108 committed:
chore: 불필요한 부분 제거

- 사용하지 않는 import제거
- retrospect에서 validate 함수 제거
- json property로 필드값 변경
